### PR TITLE
Add documentation for ++join

### PR DESCRIPTION
**Pull request by pkova**

### CODE REVIEW

**Comment by jfranklin9000** [comment]
`++join` should also be added to `reference/library/_index.md` (two places).

While you're there, line 124 (`++dem`) has spurious indentation.

**Comment by pkova** [comment]
Thanks @jfranklin9000, fixed.

Unsure of the title in _index.md, though. "Separate list" feels too vague, but what I chose is a bit wordy.

**Comment by rmariani** [comment]
I think it's a fine choice. Thanks!
